src/main/java/org/eclipse/yasson/Jsonb.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 and Eclipse Distribution License v. 1.0
+ * which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ ******************************************************************************/
+package org.eclipse.yasson;
+
+import javax.json.bind.JsonbException;
+import javax.json.stream.JsonGenerator;
+import javax.json.stream.JsonParser;
+import java.lang.reflect.Type;
+
+/**
+ * Adds methods to Jsonb that are operating directly with {@link JsonGenerator} or {@link JsonParser} types.
+ * <p>
+ * {@link javax.json.spi.JsonProvider} operates on top of the
+ * {@link java.io.InputStream} / {@link java.io.OutputStream} or {@link java.io.Reader} / {@link java.io.Writer}
+ * and creates generator / parser instances during runtime.
+ * </p>
+ * <p>
+ * This interface accepts instantiated generators and parsers with different input / output sources.
+ * </p>
+ */
+public interface Jsonb extends javax.json.bind.Jsonb {
```

Comment:
Lets name this something besides `Jsonb`, because often these classes will be imported together, and will require use of FQCN whenever used. Additionally, when users are writing code their IDEs will auto-import when there is only one option on the classpath, but now there will be two options for importing `Jsonb` which will ruin auto-import.
How about `YassonJsonb` or `ExtendedJsonb` instead?

Reply:
Hmm, I did this intentionally to make users aware of alternative and if this is eventionally introduced into API later (see linked bug), Yasson will just delete this ifc and users only change import. But you're right with FQCNs. Not sure about what the name should be, suggested names doesn't tell much about ifc intention. What about something like `ExternalJsonpJsonb` or `ExternalJsonpBinding` standing for "Jsonb for externally managed jsonp"?

Reply:
Actually, given that new interface extends the one from the API wouldn't users import one or the other, not both of them together? Most of the time they will be importing one from the API without importing new, but if they need new methods they can just use new one aren't they? Than the FCQN problem shouldn't arise.

Reply:
I agree, a user wouldn't need to import both.  Can we merge this?

Reply:
I don't think users should use one or the other, they should only use the Yasson-specific class where needed, and stick with spec-standard everywhere they possibly can.

Regardless of whether users use one or both, breaking the single-option auto-import golden path in IDEs is enough of an annoyance to warrant renaming the class to something else.

Reply:
of the two options you suggest, I prefer `ExternalJsonpBinding`

> Hmm, I did this intentionally to make users aware of alternative and if this is eventionally introduced into API later (see linked bug), Yasson will just delete this ifc and users only change import.

Assuming this gets rolled into JSON-B spec, the best we could do is collapse this class down to an interface that extended the spec `Jsonb` and adds no extra methods. We shouldn't ever delete it entirely, because that would be a breaking change for users.

Reply:
I don't like ExternalJsonpBinding.  I understand that the intent of the word "External" is to identify this interface as providing methods which do not use JsonProvider.  However, this would unnecessarily limit this interface to this specific purpose.  Its purpose should be to expose methods publicly which are not part of the standard.  You wouldn't want a new interface each time you want to add an extension method. Also the qualification of "Jsonp" doesn't distinguish this in a useful way.  JsonProvider is also part of Jsonp.   

If you rename the interface, I would suggest either:

 **JsonBinding**  - this is the same name as the internal implementation class but public users can't import the implementation since its package is not exported by the module. 

**YassonJsonb** - Qualifies this as a Yasson-specific version of Jsonb.

Reply:
+1 for `YassonJsonb`.  Also, I'm fine with anything that includes (Yasson OR Extended) AND Jsonb

Reply:
Renamed to `YassonJsonb`.

Reply:
@aguibert 

> Assuming this gets rolled into JSON-B spec, the best we could do is collapse this class down to an interface that extended the spec Jsonb and adds no extra methods. We shouldn't ever delete it entirely, because that would be a breaking change for users.

Depends if that interface is ever supposed to have other methods than those two. Should it be there empty forever, I would rather drop backwards compatibility in this case to avoid confusion and awkward javadocs.